torch/utils/data/gen_pyi.py

```diff
@@ -177,21 +233,53 @@ def main() -> None:
-                                                     "IterDataPipe", iterDP_method_to_special_output_type)
+    iter_method_definitions = get_method_definitions(root_path=iterDP_file_path,
+                                                     files_to_exclude=iterDP_files_to_exclude,
+                                                     deprecated_files=iterDP_deprecated_files,
+                                                     default_output_type="IterDataPipe",
+                                                     method_to_special_output_type=iterDP_method_to_special_output_type)
+
+    torchdata_iterDP_dir_path = "torchdata/datapipes/iter/"
+    file_contents = get_torch_data_files(torchdata_iterDP_dir_path)
+    torchdata_iterDP_to_exclude = {"__init__.py"}
+    torchdata_iterDP_method_to_special_output_type: Dict[str, str] = {"end_caching": "IterDataPipe"}
+    torchdata_iterDP_definitions = get_method_definitions(root_path="",
+                                                          file_contents=file_contents,
+                                                          files_to_exclude=torchdata_iterDP_to_exclude,
+                                                          deprecated_files=set(),
+                                                          default_output_type="IterDataPipe",
+                                                          method_to_special_output_type=torchdata_iterDP_method_to_special_output_type,
+                                                          from_torchdata=True)
+
+    iter_method_definitions += torchdata_iterDP_definitions
+
+    # TODO: When functional DataPipe is called, raise an error to warn users that they need to import TorchData
+
     mapDP_file_path: str = "datapipes/map"
     mapDP_files_to_exclude: Set[str] = {"__init__.py", "utils.py"}
     mapDP_deprecated_files: Set[str] = set()
     mapDP_method_to_special_output_type: Dict[str, str] = {}
 
-    map_method_definitions = get_method_definitions(mapDP_file_path, mapDP_files_to_exclude, mapDP_deprecated_files,
-                                                    "MapDataPipe", mapDP_method_to_special_output_type)
+    map_method_definitions = get_method_definitions(root_path=mapDP_file_path,
+                                                    files_to_exclude=mapDP_files_to_exclude,
+                                                    deprecated_files=mapDP_deprecated_files,
+                                                    default_output_type="MapDataPipe",
+                                                    method_to_special_output_type=mapDP_method_to_special_output_type)
+
+    # TODO: Get the definitions from the TorchData Map DPs
+
+    # TODO: Test if this causes an error if 'torchdata' isn't installed
+    torch_data_import_statements = ["from torchdata.datapipes.iter.util.extractor import CompressionType"]
+    torch_data_imports = ["try:"] +\
```

Comment:
Injecting these `try ... except ...` statements into the `.pyi` file seems too much of a hack to me. Any better suggestion?